bgzip.c

```diff
@@ -198,6 +203,9 @@ int main(int argc, char **argv)
 
         if ( index ) bgzf_index_build_init(fp);
         buffer = malloc(WINDOW_SIZE);
+#ifdef _WIN32
+        _setmode(f_src, O_BINARY);
+#endif
```

Comment:
I thought we decided that hopen musn't do the ```O_BINARY``` setmode thing itself, indeed I recall you squashing a PR on that very reason.  Plus again we'd be changing bgzip on Unix in order to make it run on Windows, which comes with potential bugs and risks.  I'm happier with a windows-only change personally.

The change was made for a reason (it fixes bugs).  If your view is that bgzip must NOT open everything as binary then we need to change the tool further still, adding a command line argument to tell it whether or not to treat stdin as text or binary.  Yet more complications.